crates/codegen/parser/runtime/src/napi_interface/templates/ast_selectors.rs.jinja2

```diff
@@ -253,28 +249,26 @@ pub fn select_separated(
 //
 
 struct Selector {
-    env: Env,
```

Comment:
The implicit return type conversion already use the glue generated by napi-rs (`ToNapiValue` et al.), which threads the environment anyway, so this has an added benefit of not having to do that ourselves.

Reply:
I wonder if we can also do the same with `QueryResult::bindings`, by using something like `HashMap<String, Vec<cursor.Cursor>>` instead of the untyped object?
That is the only remaining usage of `Env` I was able to find.

Reply:
It's unfortunately not possible to nest our values inside a NAPI-exposed struct due to the way ownership works Rust<>NAPI. Returning a value from the function works because it uses `ToNapiValue`, but the NAPI-exposed object members have to be `NapiValue`, which is implemented only for the JS built-ins and some of the std types, but we cannot have owned instances inside one another, see https://github.com/napi-rs/napi-rs/issues/1402.

So while

```rust
HashMap<String, Vec<Cursor>>
```
doesn't work, this does:
```rust
HashMap<String, Vec<ClassInstance<Cursor>>>
```

but we have to construct it ourselves with `Cursor(...).into_instance(env)` in the Rust->JS conversion code, so `Env` has to stay in this case.

I've done this in https://github.com/NomicFoundation/slang/pull/920/commits/02dab516e2e4b488cb48e5d04fcec5a6c1ab344a since it seemed better than having an untyped `JsObject` to better reflect how it's used.